redwood-treehouse-lazylayout-view/src/main/kotlin/app/cash/redwood/treehouse/lazylayout/view/ViewLazyList.kt

```diff
@@ -15,47 +15,60 @@
  */
 package app.cash.redwood.treehouse.lazylayout.view
 
-import android.view.Gravity
 import android.view.View
 import android.view.ViewGroup
 import android.view.ViewGroup.LayoutParams.MATCH_PARENT
 import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
 import android.widget.FrameLayout
-import androidx.paging.Pager
-import androidx.paging.PagingConfig
-import androidx.paging.PagingDataAdapter
-import androidx.paging.PagingSource
-import androidx.paging.PagingState
-import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 import app.cash.redwood.LayoutModifier
-import app.cash.redwood.treehouse.AppService
-import app.cash.redwood.treehouse.Content
-import app.cash.redwood.treehouse.HostConfiguration
-import app.cash.redwood.treehouse.TreehouseApp
-import app.cash.redwood.treehouse.TreehouseView.WidgetSystem
-import app.cash.redwood.treehouse.TreehouseWidgetView
-import app.cash.redwood.treehouse.lazylayout.api.LazyListInterval
 import app.cash.redwood.treehouse.lazylayout.widget.LazyList
+import app.cash.redwood.widget.MutableListChildren
+import app.cash.redwood.widget.Widget
 import kotlinx.coroutines.MainScope
 import kotlinx.coroutines.cancel
-import kotlinx.coroutines.flow.collectLatest
-import kotlinx.coroutines.launch
 
-internal class ViewLazyList<A : AppService>(
-  private val treehouseApp: TreehouseApp<A>,
-  widgetSystem: WidgetSystem,
+internal class Items<VH : RecyclerView.ViewHolder>(
+  private val adapter: RecyclerView.Adapter<VH>,
+) : Widget.Children<View> {
+  private val _widgets = MutableListChildren<View>()
+  val widgets: List<Widget<View>> get() = _widgets
+
+  override fun insert(index: Int, widget: Widget<View>) {
+    _widgets.insert(index, widget)
+    adapter.notifyItemInserted(index)
+  }
+
+  override fun move(fromIndex: Int, toIndex: Int, count: Int) {
+    _widgets.move(fromIndex, toIndex, count)
+    check(count == 1)
+    adapter.notifyItemMoved(fromIndex, toIndex) // TODO Support arbitrary count.
```

Comment:
Probably have to notify a range change on >1. I _think_ it would be something like:
```kotlin
val lowerBound = minOf(fromIndex, toIndex)
val upperBound = maxOf(fromIndex, toIndex) + count
```

Reply:
I'm not too sure what the `notify*` method I should be calling. Ideally I'd want something like `adapter.notifyItemRangeMoved(fromPosition, toPosition, count)`, but no such method exists. Frustratingly, stepping through the implementation of `RecyclerView.Adapter#notifyItemMoved` shows that it's invoking `RecyclerView.AdapterDataObserver#onItemRangeMoved(fromPosition, toPosition, itemCount)` with `itemCount` hardcoded to 1.

Reply:
Hah! File a feature request / send a PR?

Reply:
[Filed!](https://issuetracker.google.com/issues/281929801)